src/test/ui/error-codes/E0010-teach.stderr

```diff
@@ -6,6 +6,16 @@ LL | const CON : Box<i32> = box 0; //~ ERROR E0010
    |
    = note: The value of statics and constants must be known at compile time, and they live for the entire lifetime of a program. Creating a boxed value allocates memory on the heap at runtime, and therefore cannot be done at compile time.
 
-error: aborting due to previous error
+error[E0019]: constant contains unimplemented expression type
+  --> $DIR/E0010-teach.rs:6:28
+   |
+LL | const CON : Box<i32> = box 0; //~ ERROR E0010
+   |                            ^
+   |
+   = note: A function call isn't allowed in the const's initialization expression because the expression's value must be known at compile-time.
```

Comment:
This reads hilariously given that from the user's perspective, there's no function call being made in `box 0`...
If not too troublesome, improve the diagnostic (we can also do this another time...)?

Reply:
yea we have this everywhere, this is nothing new to this PR

Reply:
Fair -- no need to fix this here, but if someone does fix it at some future juncture it would be nice.